### PR TITLE
Added OFDM support 

### PR DESCRIPTION
**Pull request by sarafalleni05**
Added: 
OFDM phy module 
OFDM interference module
OFDM Aloha compatible with OFDM physical layer 
tcl test file as usage example
msg_displayer class for convenience 
modifications to previous classes to better support OFDM doings

### CODE REVIEW

**Review by fedefava86 — CHANGES_REQUESTED** [review]

File: DESERT_Framework/DESERT/samples/desert_samples/test_ofdmaloha_desert_integration.tcl [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+        puts "- the sixth one is the Packet Size"
+        puts "example: ns test_uwcbr_macalohaofdm.tcl 5 fullBandexp.csv 5 BPSK 1 1536"
+        puts "Please try again."
+        return
+    } else {
+        set opt(cbr_period)    [lindex $argv 0]
+        set opt(filename)      [lindex $argv 1]
+        set opt(nn)            [lindex $argv 2] 
+        set opt(subCarSize)    [expr $opt(bw)/$opt(subCarrier) ] 
+        set opt(modulation)     [lindex $argv 4]
```
Comment (fedefava86) [inline DESERT_Framework/DESERT/samples/desert_samples/test_ofdmaloha_desert_integration.tcl:160]: Index of argv is broken, should be 3
```suggestion
        set opt(modulation)     [lindex $argv 3]
```

File: DESERT_Framework/DESERT/samples/desert_samples/test_ofdmaloha_desert_integration.tcl [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+        set opt(AckInBand)     [lindex $argv 5]
```
Comment (fedefava86) [inline DESERT_Framework/DESERT/samples/desert_samples/test_ofdmaloha_desert_integration.tcl:161]:
```suggestion
        set opt(AckInBand)     [lindex $argv 4]
```

File: DESERT_Framework/DESERT/samples/desert_samples/test_ofdmaloha_desert_integration.tcl [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+        set opt(pktsize)        [lindex $argv 6]
```
Comment (fedefava86) [inline DESERT_Framework/DESERT/samples/desert_samples/test_ofdmaloha_desert_integration.tcl:162]:
```suggestion
        set opt(pktsize)        [lindex $argv 5]
```

**Review by fedefava86 — CHANGES_REQUESTED** [review]

File: DESERT_Framework/DESERT/interference/uwinterferenceofdm/uwinterferenceofdm.cpp [hunk 4]
```diff
@@ -0,0 +1,510 @@
...
+				it = power_list.erase(it); // Side effect: it++
+			}
+			else
+				break;
+		}
+	}
+
+	if (power_list.empty()) {
+
+		std::cerr << "uwinterference::removeFromInterference, "
```
[the remark this answers is not in the document]
Comment (sarafalleni05) [reply DESERT_Framework/DESERT/interference/uwinterferenceofdm/uwinterferenceofdm.cpp:241]: I will return in that case. However this code is written in the same way in the original interference module, maybe you also want to change it over there.

[the remark this answers is not in the document]
Comment (fedefava86) [reply DESERT_Framework/DESERT/interference/uwinterferenceofdm/uwinterferenceofdm.cpp:241]: Yes, definitely. So we have a potential crash too, good to know. Thank you!

**Review by fedefava86 — APPROVED** [review]

**Pull request merged**